azure-iot-device/tests/common/pipeline/test_pipeline_stages_base.py

```diff
@@ -252,6 +252,14 @@ def stage(self, mocker, cls_type, init_kwargs):
 )
 
 
+only_disconnected = [pytest.param(False, id="Pipeline Disconnected")]
```

Comment:
nit: don't think this is great for test readability... I can see how it helps when writing the tests.

Reply:
I agree with Yoseph, I did some stuff like this before and it ended up being more of a pain than it was worth to do single parametrization. Best off just breaking down the test cases IMO

---
In reply to: [387896996](https://github.com/Azure/azure-iot-sdk-python/pull/476#discussion_r387896996) [](ancestors = 387896996)

Reply:
OK, that makes sense.  I was just trying to avoid repeated code, plus I was having trouble phrasing the it.  e.g.  it("Sends down another connect operation if the pipeline was not connected before the operation, then successfully connected, then the op failed and the pipeline was no longer connected after the op failed")

---
In reply to: [388041552](https://github.com/Azure/azure-iot-sdk-python/pull/476#discussion_r388041552) [](ancestors = 388041552,387896996)